pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -288,7 +324,6 @@ func (r *ReconcileBareMetalHost) getBMCCredentials(request reconcile.Request, in
 	if secretKey.Namespace == "" {
```

Comment:
I think we should require that the secret be in the same namespace as the host object. Aside from being the likely case and a sensible way to keep related data together, OwnerRefs cannot traverse namespaces.

Reply:
Should I not use a reference then, and just ask for the secret name in the spec?

Reply:
I think just the name would be fine, and it would match what I've seen other operators do.

Reply:
Done.